pkg/trace/api/otlp.go

```diff
@@ -517,6 +517,12 @@ func (o *OTLPReceiver) convertSpan(rattr map[string]string, lib pcommon.Instrume
 	if in.Events().Len() > 0 {
 		transform.SetMetaOTLP(span, "events", transform.MarshalEvents(in.Events()))
 	}
+	for i := range in.Events().Len() {
```

Comment:
Were already iterating through events/ checking for exceptions [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/transform/transform.go#L333), may be more efficient to add `span.Meta["_dd.span_events.has_exception"] = "true"` below that line.

It also seems like `Status2Error` gets called in both paths you changed, so moving to there would only require adding the logic in 1 place.

Reply:
From what I can see, there is a check in `Status2Error` [here](https://github.com/DataDog/datadog-agent/blob/c079441bf0d6b2fc6eeaabb95d9fae69a9da4e34/pkg/trace/transform/transform.go#L325) so that it is only applied to error spans.

The point being to consider non error spans that do contain exception span events in the error sampler (in addition to error spans), I don't think I can move the tag setting there unfortunately.

Reply:
> The point being to consider non error spans that do contain exception span events in the error sampler

Ah, wasn't aware of this. Agreed you can't add to `Status2Error` then. May be good to still create a func to avoid duplication but is fine as is too, approving.

Reply:
Moved into its own func in https://github.com/DataDog/datadog-agent/pull/30065/commits/7f8a8e5fc5efccf9bc22237990d5d91dc5616f30, cleaner indeed!